flow/datatypes/numeric.go

```diff
@@ -46,12 +53,20 @@ func (SnowflakeNumericCompatibility) MaxScale() int16 {
 	return 37
 }
 
-func (SnowflakeNumericCompatibility) DefaultPrecisionAndScale() (int16, int16) {
-	return PeerDBSnowflakePrecision, PeerDBSnowflakeScale
+func (s SnowflakeNumericCompatibility) DefaultPrecisionAndScale() (int16, int16) {
+	return s.MaxPrecision(), PeerDBSnowflakeScale
+}
+
+func (s SnowflakeNumericCompatibility) IsValidPrecisionAndScale(precision, scale int16) bool {
+	return s.IsValidPrecision(precision) && s.IsValidScale(scale) && scale < precision
 }
 
-func (SnowflakeNumericCompatibility) IsValidPrecisionAndScale(precision, scale int16) bool {
-	return precision > 0 && precision <= 38 && scale < precision
+func (s SnowflakeNumericCompatibility) IsValidPrecision(precision int16) bool {
```

Comment:
when these methods all have same result they can instead take be free functions taking `WarehouseNumericCompatibility`